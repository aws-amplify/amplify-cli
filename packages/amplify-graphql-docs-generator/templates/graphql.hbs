# this is an auto generated file. This will be overwritten
{{#each operations }}
  {{> renderOp }}
{{/each}}
{{#each fragments }}
  {{> renderExternalFragment }}
{{/each}}