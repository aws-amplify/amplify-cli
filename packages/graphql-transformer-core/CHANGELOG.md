# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="1.0.28"></a>
## [1.0.28](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.28-beta.0...graphql-transformer-core@1.0.28) (2018-10-18)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.28-beta.0"></a>
## [1.0.28-beta.0](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.12...graphql-transformer-core@1.0.28-beta.0) (2018-10-12)


### Bug Fixes

* **graphql-transformer-core:** Fix Readme.md docs for the `[@auth](https://github.com/auth)` directive ([db6ff7a](https://github.com/aws-amplify/amplify-cli/commit/db6ff7a))




<a name="1.0.12"></a>
## [1.0.12](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.11...graphql-transformer-core@1.0.12) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.11"></a>
## [1.0.11](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.10...graphql-transformer-core@1.0.11) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.10"></a>
## [1.0.10](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.9...graphql-transformer-core@1.0.10) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.9"></a>
## [1.0.9](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.8...graphql-transformer-core@1.0.9) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.8"></a>
## [1.0.8](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.7...graphql-transformer-core@1.0.8) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.7"></a>
## [1.0.7](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.6...graphql-transformer-core@1.0.7) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.6"></a>
## [1.0.6](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.5...graphql-transformer-core@1.0.6) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.5"></a>
## [1.0.5](https://github.com/aws-amplify/amplify-cli/compare/graphql-transformer-core@1.0.4...graphql-transformer-core@1.0.5) (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.4"></a>
## 1.0.4 (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.3"></a>
## 1.0.3 (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.2"></a>
## 1.0.2 (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core

<a name="1.0.1"></a>
## 1.0.1 (2018-08-23)




**Note:** Version bump only for package graphql-transformer-core
