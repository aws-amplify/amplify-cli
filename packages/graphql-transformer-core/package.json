{
  "name": "graphql-transformer-core",
  "version": "6.19.4",
  "description": "A framework to transform from GraphQL SDL to AWS CloudFormation.",
  "repository": {
    "type": "git",
    "url": "https://github.com/aws-amplify/amplify-cli.git",
    "directory": "packages/graphql-transformer-core"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "keywords": [
    "graphql",
    "cloudformation",
    "aws"
  ],
  "scripts": {
    "test": "jest",
    "test-ci": "jest --ci -i",
    "build": "tsc",
    "watch": "tsc -w",
    "clean": "rimraf ./lib"
  },
  "dependencies": {
    "cloudform-types": "^4.2.0",
    "deep-diff": "^1.0.2",
    "fs-extra": "^8.1.0",
    "glob": "^7.1.6",
    "graphql": "^14.5.8",
    "graphql-transformer-common": "4.17.4"
  },
  "devDependencies": {
    "@types/fs-extra": "^8.0.1",
    "@types/node": "^10.17.13"
  },
  "jest": {
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    },
    "testURL": "http://localhost",
    "testRegex": "(src/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "collectCoverage": true
  }
}
