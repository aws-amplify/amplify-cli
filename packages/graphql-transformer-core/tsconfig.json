{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "strict": false, // TODO enable
    "rootDir": "src",
    "outDir": "lib"
  },
  "references": [
    {"path": "../graphql-transformer-common"}
  ]
}
