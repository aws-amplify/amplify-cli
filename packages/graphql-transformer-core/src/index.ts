import './polyfills/Object.assign';
import { print } from 'graphql';
import { TransformerContext } from './TransformerContext';
import { Transformer } from './Transformer';
import { ITransformer } from './ITransformer';
import { GraphQLTransform } from './GraphQLTransform';
import { collectDirectiveNames, collectDirectivesByTypeNames } from './collectDirectives';
import { stripDirectives } from './stripDirectives';
import { DeploymentResources } from './DeploymentResources';
import {
  buildProject as buildAPIProject,
  uploadDeployment as uploadAPIProject,
  migrateAPIProject,
  revertAPIMigration,
} from './util/amplifyUtils';
import {
  readSchema as readProjectSchema,
  loadProject as readProjectConfiguration,
  loadConfig as readTransformerConfiguration,
  writeConfig as writeTransformerConfiguration,
  TRANSFORM_CONFIG_FILE_NAME,
  TRANSFORM_BASE_VERSION,
  TRANSFORM_CURRENT_VERSION,
  TransformConfig,
  SyncConfig,
} from './util/transformConfig';
import { EXTRA_DIRECTIVES_DOCUMENT } from './validation';
export * from './errors';
export * from './util';

/**
 * Returns the extra set of directives that are supported by AppSync service
 */
export function getAppSyncServiceExtraDirectives(): string {
  return print(EXTRA_DIRECTIVES_DOCUMENT);
}

export {
  GraphQLTransform,
  TransformConfig,
  TransformerContext,
  Transformer,
  ITransformer,
  collectDirectiveNames,
  collectDirectivesByTypeNames,
  stripDirectives,
  buildAPIProject,
  migrateAPIProject,
  uploadAPIProject,
  readProjectSchema,
  readProjectConfiguration,
  readTransformerConfiguration,
  writeTransformerConfiguration,
  revertAPIMigration,
  TRANSFORM_CONFIG_FILE_NAME,
  TRANSFORM_BASE_VERSION,
  TRANSFORM_CURRENT_VERSION,
  SyncConfig,
  DeploymentResources,
};
