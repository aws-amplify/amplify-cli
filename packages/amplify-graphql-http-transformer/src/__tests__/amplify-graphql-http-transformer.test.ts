'use strict';
import { GraphQLTransform } from '@aws-amplify/graphql-transformer-core';
import { anything, countResources, expect as cdkExpect, haveResource } from '@aws-cdk/assert';
import { parse } from 'graphql';
import { HttpTransformer } from '..';

test('generates expected VTL', () => {
  const validSchema = `
    type Comment {
      id: ID!
      content: String @http(url: "https://www.api.com/ping", headers: [{key: "X-Header", value: "X-Header-Value"}])
      contentDelete: String @http(method: DELETE, url: "https://www.api.com/ping", headers: [{key: "X-Header", value: "X-Header-ValueDelete"}])
      contentPatch: String @http(method: PATCH, url: "https://www.api.com/ping", headers: [{key: "X-Header", value: "X-Header-ValuePatch"}])
      contentPost: String @http(method: POST, url: "https://www.api.com/ping", headers: [{key: "X-Header", value: "X-Header-ValuePost"}])
      complexPut(
        id: Int!,
        title: String!,
        body: String,
        userId: Int!
      ): String @http(method: PUT, url: "https://jsonplaceholder.typicode.com/posts/:title/:id/\${ctx.source.id}", headers: [{key: "X-Header", value: "X-Header-ValuePut"}])
    }
    `;
  const transformer = new GraphQLTransform({
    transformers: [new HttpTransformer()],
  });
  const out = transformer.transform(validSchema);
  expect(out).toBeDefined();
  expect(out.stacks).toBeDefined();
  expect(out.resolvers).toMatchSnapshot();
  expect(out.pipelineFunctions).toMatchSnapshot();
  parse(out.schema);
});

test('it generates the expected resources', () => {
  const validSchema = `
    type Comment {
      id: ID!
      content: String @http(method: POST, url: "http://www.api.com/ping")
      content2: String @http(method: PUT, url: "http://www.api.com/ping")
      more: String @http(url: "http://api.com/ping/me/2")
      evenMore: String @http(method: DELETE, url: "http://www.google.com/query/id")
      stillMore: String @http(method: PATCH, url: "https://www.api.com/ping/id")
    }
    `;
  const transformer = new GraphQLTransform({
    transformers: [new HttpTransformer()],
  });
  const out = transformer.transform(validSchema);
  expect(out).toBeDefined();
  expect(out.stacks).toBeDefined();
  parse(out.schema);
  const stack = out.stacks.HttpStack;
  cdkExpect(stack).to(
    haveResource('AWS::IAM::Role', {
      AssumeRolePolicyDocument: {
        Statement: [
          {
            Action: 'sts:AssumeRole',
            Effect: 'Allow',
            Principal: {
              Service: 'appsync.amazonaws.com',
            },
          },
        ],
        Version: '2012-10-17',
      },
    }),
  );
  cdkExpect(stack).to(countResources('AWS::AppSync::DataSource', 4));
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      ApiId: { Ref: anything() },
      Name: 'httpwwwapicomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: 'http://www.api.com',
      },
      ServiceRoleArn: {
        'Fn::GetAtt': [anything(), 'Arn'],
      },
    }),
  );
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      ApiId: { Ref: anything() },
      Name: 'httpapicomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: 'http://api.com',
      },
      ServiceRoleArn: {
        'Fn::GetAtt': [anything(), 'Arn'],
      },
    }),
  );
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      ApiId: { Ref: anything() },
      Name: 'httpwwwgooglecomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: 'http://www.google.com',
      },
      ServiceRoleArn: {
        'Fn::GetAtt': [anything(), 'Arn'],
      },
    }),
  );
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      ApiId: { Ref: anything() },
      Name: 'httpswwwapicomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: 'https://www.api.com',
      },
      ServiceRoleArn: {
        'Fn::GetAtt': [anything(), 'Arn'],
      },
    }),
  );
  cdkExpect(stack).to(countResources('AWS::AppSync::Resolver', 5));
  cdkExpect(stack).to(countResources('AWS::AppSync::FunctionConfiguration', 5));
  expect(stack.Resources!.CommentcontentResolver).toBeTruthy();
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::Resolver', {
      ApiId: { Ref: anything() },
      FieldName: 'content',
      TypeName: 'Comment',
      Kind: 'PIPELINE',
    }),
  );
  expect(stack.Resources!.Commentcontent2Resolver).toBeTruthy();
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::Resolver', {
      ApiId: { Ref: anything() },
      FieldName: 'content2',
      TypeName: 'Comment',
      Kind: 'PIPELINE',
    }),
  );
  expect(stack.Resources!.CommentmoreResolver).toBeTruthy();
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::Resolver', {
      ApiId: { Ref: anything() },
      FieldName: 'more',
      TypeName: 'Comment',
      Kind: 'PIPELINE',
    }),
  );
  expect(stack.Resources!.CommentevenMoreResolver).toBeTruthy();
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::Resolver', {
      ApiId: { Ref: anything() },
      FieldName: 'evenMore',
      TypeName: 'Comment',
      Kind: 'PIPELINE',
    }),
  );
  expect(stack.Resources!.CommentstillMoreResolver).toBeTruthy();
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::Resolver', {
      ApiId: { Ref: anything() },
      FieldName: 'stillMore',
      TypeName: 'Comment',
      Kind: 'PIPELINE',
    }),
  );
});

test('URL params happy path', () => {
  const validSchema = `
    type Comment {
      id: ID!
      title: String
      complex: CompObj @http(method: GET, url: "https://jsonplaceholder.typicode.com/posts/1")
      complexAgain: CompObj @http(url: "https://jsonplaceholder.typicode.com/posts/2")
      complexPost(
        id: Int,
        title: String,
        body: String,
        userId: Int
      ): CompObj @http(method: POST, url: "https://jsonplaceholder.typicode.com/posts")
      complexPut(
        id: Int!,
        title: String!,
        body: String,
        userId: Int!
      ): CompObj @http(method: PUT, url: "https://jsonplaceholder.typicode.com/posts/:title/:id")
      deleter: String @http(method: DELETE, url: "https://jsonplaceholder.typicode.com/posts/3")
      complexGet(
        id: Int!
      ): CompObj @http(url: "https://jsonplaceholder.typicode.com/posts/:id")
      complexGet2 (
        id: Int!,
        title: String!,
        userId: Int!
      ): CompObj @http(url: "https://jsonplaceholder.typicode.com/posts/:title/:id")
    }
    type CompObj {
      userId: Int
      id: Int
      title: String
      body: String
    }
    `;
  const transformer = new GraphQLTransform({
    transformers: [new HttpTransformer()],
  });
  const out = transformer.transform(validSchema);
  expect(out).toBeDefined();
  expect(out.stacks).toBeDefined();
  parse(out.schema);
  const stack = out.stacks.HttpStack;
  cdkExpect(stack).to(countResources('AWS::AppSync::DataSource', 1));
  cdkExpect(stack).to(countResources('AWS::AppSync::Resolver', 7));
  expect(stack.Resources!.CommentcomplexResolver).toBeTruthy();
  expect(stack.Resources!.CommentcomplexAgainResolver).toBeTruthy();
  expect(stack.Resources!.CommentcomplexPostResolver).toBeTruthy();
  expect(stack.Resources!.CommentcomplexPutResolver).toBeTruthy();
  expect(stack.Resources!.CommentdeleterResolver).toBeTruthy();
  expect(stack.Resources!.CommentcomplexGetResolver).toBeTruthy();
  expect(stack.Resources!.CommentcomplexGet2Resolver).toBeTruthy();
});

test('it throws an error when missing protocol in URL argument', () => {
  const validSchema = `
    type Comment {
      id: ID!
      content: String @http(method: POST, url: "www.api.com/ping")
    }
    `;
  const transformer = new GraphQLTransform({
    transformers: [new HttpTransformer()],
  });

  expect(() => {
    transformer.transform(validSchema);
  }).toThrow('@http directive at location 56 requires a url parameter that begins with http:// or https://.');
});

test('env on the URI path', () => {
  const validSchema = `
    type Comment {
      id: ID!
      content: String @http(method: POST, url: "http://www.api.com/ping\${env}")
    }
  `;
  const transformer = new GraphQLTransform({
    transformers: [new HttpTransformer()],
  });
  const out = transformer.transform(validSchema);
  expect(out).toBeDefined();
  expect(out.stacks).toBeDefined();
  parse(out.schema);
  const stack = out.stacks.HttpStack;
  const functionId = stack.Resources!.CommentcontentResolver.Properties.PipelineConfig.Functions[0]['Fn::GetAtt'][0];
  const reqTemplate = stack.Resources![functionId].Properties.RequestMappingTemplate;
  expect(reqTemplate['Fn::Sub']).toBeTruthy();
  expect(reqTemplate['Fn::Sub'][0]).toMatch('"resourcePath": "/ping${env}"');
  expect(reqTemplate['Fn::Sub'][1].env.Ref).toBeTruthy();
});

test('env on the hostname', () => {
  const validSchema = `
    type Comment {
      id: ID!
      content: String @http(method: POST, url: "http://\${env}www.api.com/ping")
      content2: String @http(method: PUT, url: "http://\${env}www.api.com/ping")
      more: String @http(url: "http://\${env}api.com/ping/me/2")
      evenMore: String @http(method: DELETE, url: "http://\${env}www.google.com/query/id")
      stillMore: String @http(method: PATCH, url: "https://\${env}www.api.com/ping/id")
    }
  `;
  const transformer = new GraphQLTransform({
    transformers: [new HttpTransformer()],
  });
  const out = transformer.transform(validSchema);
  expect(out).toBeDefined();
  expect(out.stacks).toBeDefined();
  parse(out.schema);
  const stack = out.stacks.HttpStack;
  cdkExpect(stack).to(countResources('AWS::AppSync::DataSource', 4));
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      Name: 'httpenvwwwapicomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: {
          'Fn::Sub': [
            'http://${env}www.api.com',
            {
              env: {
                Ref: anything(),
              },
            },
          ],
        },
      },
    }),
  );
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      Name: 'httpenvapicomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: {
          'Fn::Sub': [
            'http://${env}api.com',
            {
              env: {
                Ref: anything(),
              },
            },
          ],
        },
      },
    }),
  );
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      Name: 'httpenvwwwgooglecomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: {
          'Fn::Sub': [
            'http://${env}www.google.com',
            {
              env: {
                Ref: anything(),
              },
            },
          ],
        },
      },
    }),
  );
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      Name: 'httpsenvwwwapicomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: {
          'Fn::Sub': [
            'https://${env}www.api.com',
            {
              env: {
                Ref: anything(),
              },
            },
          ],
        },
      },
    }),
  );
});

test('aws_region on the URI path', () => {
  const validSchema = `
    type Comment {
      id: ID!
      content: String @http(method: POST, url: "http://www.api.com/ping\${aws_region}")
    }
  `;
  const transformer = new GraphQLTransform({
    transformers: [new HttpTransformer()],
  });
  const out = transformer.transform(validSchema);
  expect(out).toBeDefined();
  expect(out.stacks).toBeDefined();
  parse(out.schema);
  const stack = out.stacks.HttpStack;
  const functionId = stack.Resources!.CommentcontentResolver.Properties.PipelineConfig.Functions[0]['Fn::GetAtt'][0];
  const reqTemplate = stack.Resources![functionId].Properties.RequestMappingTemplate;
  expect(reqTemplate['Fn::Sub']).toBeTruthy();
  expect(reqTemplate['Fn::Sub'][0]).toMatch('"resourcePath": "/ping${aws_region}"');
  expect(reqTemplate['Fn::Sub'][1].aws_region.Ref).toBeTruthy();
});

test('aws_region on the hostname', () => {
  const validSchema = `
    type Comment {
      id: ID!
      content: String @http(method: POST, url: "http://\${aws_region}www.api.com/ping")
      content2: String @http(method: PUT, url: "http://\${aws_region}www.api.com/ping")
      more: String @http(url: "http://\${aws_region}api.com/ping/me/2")
      evenMore: String @http(method: DELETE, url: "http://\${aws_region}www.google.com/query/id")
      stillMore: String @http(method: PATCH, url: "https://\${aws_region}www.api.com/ping/id")
    }
  `;
  const transformer = new GraphQLTransform({
    transformers: [new HttpTransformer()],
  });
  const out = transformer.transform(validSchema);
  expect(out).toBeDefined();
  expect(out.stacks).toBeDefined();
  parse(out.schema);
  const stack = out.stacks.HttpStack;
  cdkExpect(stack).to(countResources('AWS::AppSync::DataSource', 4));
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      Name: 'httpaws_regionwwwapicomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: {
          'Fn::Sub': [
            'http://${aws_region}www.api.com',
            {
              aws_region: {
                Ref: anything(),
              },
            },
          ],
        },
      },
    }),
  );
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      Name: 'httpaws_regionapicomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: {
          'Fn::Sub': [
            'http://${aws_region}api.com',
            {
              aws_region: {
                Ref: anything(),
              },
            },
          ],
        },
      },
    }),
  );
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      Name: 'httpaws_regionwwwgooglecomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: {
          'Fn::Sub': [
            'http://${aws_region}www.google.com',
            {
              aws_region: {
                Ref: anything(),
              },
            },
          ],
        },
      },
    }),
  );
  cdkExpect(stack).to(
    haveResource('AWS::AppSync::DataSource', {
      Name: 'httpsaws_regionwwwapicomDataSource',
      Type: 'HTTP',
      HttpConfig: {
        Endpoint: {
          'Fn::Sub': [
            'https://${aws_region}www.api.com',
            {
              aws_region: {
                Ref: anything(),
              },
            },
          ],
        },
      },
    }),
  );
});
