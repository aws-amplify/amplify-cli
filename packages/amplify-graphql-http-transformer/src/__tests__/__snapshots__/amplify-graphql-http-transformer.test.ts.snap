// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates expected VTL 1`] = `
Object {
  "Comment.complexPut.DataResolver.req.vtl": "## [Start] Create request. **
## START: Manually checking that all non-null arguments are provided either in the query or the body **
#if( (!$ctx.args.body.userId && !$ctx.args.query.userId) )
$util.error(\\"An argument you marked as Non-Null is not present in the query nor the body of your request.\\"))
#end
## END: Manually checking that all non-null arguments are provided either in the query or the body **
#set( $headers = $utils.http.copyHeaders($ctx.request.headers) )
$util.qr($headers.put(\\"accept-encoding\\", \\"application/json\\"))
$util.qr($headers.put(\\"Content-Type\\", \\"application/json\\"))
$util.qr($headers.put(\\"X-Header\\", \\"X-Header-ValuePut\\"))
{
  \\"version\\": \\"2018-05-29\\",
  \\"method\\": \\"PUT\\",
  \\"resourcePath\\": \\"/posts/\${ctx.args.params.title}/\${ctx.args.params.id}/\${ctx.source.id}\\",
  \\"params\\": {
      \\"headers\\": $util.toJson($headers),
      \\"query\\": $util.toJson($ctx.args.query),
      \\"body\\": $util.toJson($ctx.args.body)
  }
}
## [End] Create request. **",
  "Comment.complexPut.DataResolver.res.vtl": "## [Start] Process response. **
#if( $ctx.result.statusCode == 200 || $ctx.result.statusCode == 201 )
  #if( $ctx.result.headers.get(\\"Content-Type\\").toLowerCase().contains(\\"xml\\") )
$utils.xml.toJsonString($ctx.result.body)
  #else
$ctx.result.body
  #end
#else
$util.qr($util.appendError($ctx.result.body, $ctx.result.statusCode))
#end
## [End] Process response. **",
  "Comment.complexPut.res.vtl": "$util.toJson($ctx.prev.result)",
  "Comment.content.DataResolver.req.vtl": "## [Start] Create request. **
#set( $headers = $utils.http.copyHeaders($ctx.request.headers) )
$util.qr($headers.put(\\"accept-encoding\\", \\"application/json\\"))
$util.qr($headers.put(\\"X-Header\\", \\"X-Header-Value\\"))
{
  \\"version\\": \\"2018-05-29\\",
  \\"method\\": \\"GET\\",
  \\"resourcePath\\": \\"/ping\\",
  \\"params\\": {
      \\"headers\\": $util.toJson($headers),
      \\"query\\": $util.toJson($ctx.args.query)
  }
}
## [End] Create request. **",
  "Comment.content.DataResolver.res.vtl": "## [Start] Process response. **
#if( $ctx.result.statusCode == 200 )
  #if( $ctx.result.headers.get(\\"Content-Type\\").toLowerCase().contains(\\"xml\\") )
$utils.xml.toJsonString($ctx.result.body)
  #else
$ctx.result.body
  #end
#else
$util.qr($util.appendError($ctx.result.body, $ctx.result.statusCode))
#end
## [End] Process response. **",
  "Comment.content.res.vtl": "$util.toJson($ctx.prev.result)",
  "Comment.contentDelete.DataResolver.req.vtl": "## [Start] Create request. **
#set( $headers = $utils.http.copyHeaders($ctx.request.headers) )
$util.qr($headers.put(\\"accept-encoding\\", \\"application/json\\"))
$util.qr($headers.put(\\"X-Header\\", \\"X-Header-ValueDelete\\"))
{
  \\"version\\": \\"2018-05-29\\",
  \\"method\\": \\"DELETE\\",
  \\"resourcePath\\": \\"/ping\\",
  \\"params\\": {
      \\"headers\\": $util.toJson($headers)
  }
}
## [End] Create request. **",
  "Comment.contentDelete.DataResolver.res.vtl": "## [Start] Process response. **
#if( $ctx.result.statusCode == 200 )
  #if( $ctx.result.headers.get(\\"Content-Type\\").toLowerCase().contains(\\"xml\\") )
$utils.xml.toJsonString($ctx.result.body)
  #else
$ctx.result.body
  #end
#else
$util.qr($util.appendError($ctx.result.body, $ctx.result.statusCode))
#end
## [End] Process response. **",
  "Comment.contentDelete.res.vtl": "$util.toJson($ctx.prev.result)",
  "Comment.contentPatch.DataResolver.req.vtl": "## [Start] Create request. **
#set( $headers = $utils.http.copyHeaders($ctx.request.headers) )
$util.qr($headers.put(\\"accept-encoding\\", \\"application/json\\"))
$util.qr($headers.put(\\"Content-Type\\", \\"application/json\\"))
$util.qr($headers.put(\\"X-Header\\", \\"X-Header-ValuePatch\\"))
{
  \\"version\\": \\"2018-05-29\\",
  \\"method\\": \\"PATCH\\",
  \\"resourcePath\\": \\"/ping\\",
  \\"params\\": {
      \\"headers\\": $util.toJson($headers),
      \\"query\\": $util.toJson($ctx.args.query),
      \\"body\\": $util.toJson($ctx.args.body)
  }
}
## [End] Create request. **",
  "Comment.contentPatch.DataResolver.res.vtl": "## [Start] Process response. **
#if( $ctx.result.statusCode == 200 || $ctx.result.statusCode == 201 )
  #if( $ctx.result.headers.get(\\"Content-Type\\").toLowerCase().contains(\\"xml\\") )
$utils.xml.toJsonString($ctx.result.body)
  #else
$ctx.result.body
  #end
#else
$util.qr($util.appendError($ctx.result.body, $ctx.result.statusCode))
#end
## [End] Process response. **",
  "Comment.contentPatch.res.vtl": "$util.toJson($ctx.prev.result)",
  "Comment.contentPost.DataResolver.req.vtl": "## [Start] Create request. **
#set( $headers = $utils.http.copyHeaders($ctx.request.headers) )
$util.qr($headers.put(\\"accept-encoding\\", \\"application/json\\"))
$util.qr($headers.put(\\"Content-Type\\", \\"application/json\\"))
$util.qr($headers.put(\\"X-Header\\", \\"X-Header-ValuePost\\"))
{
  \\"version\\": \\"2018-05-29\\",
  \\"method\\": \\"POST\\",
  \\"resourcePath\\": \\"/ping\\",
  \\"params\\": {
      \\"headers\\": $util.toJson($headers),
      \\"query\\": $util.toJson($ctx.args.query),
      \\"body\\": $util.toJson($ctx.args.body)
  }
}
## [End] Create request. **",
  "Comment.contentPost.DataResolver.res.vtl": "## [Start] Process response. **
#if( $ctx.result.statusCode == 200 || $ctx.result.statusCode == 201 )
  #if( $ctx.result.headers.get(\\"Content-Type\\").toLowerCase().contains(\\"xml\\") )
$utils.xml.toJsonString($ctx.result.body)
  #else
$ctx.result.body
  #end
#else
$util.qr($util.appendError($ctx.result.body, $ctx.result.statusCode))
#end
## [End] Process response. **",
  "Comment.contentPost.res.vtl": "$util.toJson($ctx.prev.result)",
}
`;

exports[`generates expected VTL 2`] = `Object {}`;
